webapp/src/store/tasks.ts

```diff
@@ -16,8 +16,12 @@ export const useTasksStore = defineStore('tasks', () => {
   
   // used to display loading indicator in the webapp when loading tasks
   const loading = ref(false)
```

Comment:
haa, my typing demons are coming back to haunt me!

consider using a more fun type, like `const loading = ref<'successed' | 'failed' | false>`.
no need to play with many variables, and it ensure state consistency.

Reply:
Very neat! I don't like that  `if(loading) console.log(loading)` would print 'success' or 'failed'. This is unexpected since loading is true (it should be loading)
I'm changing it to `const status = ref<'success' | 'failed' | 'loading'>` to make everything explicit